src/macos.rs

```diff
@@ -245,6 +253,18 @@ pub(crate) fn configure_dns_servers(dns_servers: &[IpAddr]) -> std::io::Result<(
     Ok(())
 }
 
+fn config_dns_servers_empty(original_gw_scope: &str) -> std::io::Result<()> {
```

Comment:
There is absolutely no need to implement the logic of this function through a shell script.